ios/platform-features/auto-layout/programmatical-constraints.workbook

```diff
@@ -0,0 +1,120 @@
+---
+uti: com.xamarin.workbook
+platforms:
+- iOS
+---
+
+### Programmatically Creating Constraints
+
+The goal of this workbook is to explain **advanced auto layout** concepts by showing how to create constraints **programmatically**.
+
+*Whenever possible, we recommend you use the iOS designer to create your contraints because you can easily visualize, edit, manage and debug them.*
+
+```csharp
+var orangeLabel = new UILabel () {
+        Text = "Orange",
+        TextColor = UIColor.White,
+        TextAlignment = UITextAlignment.Center,
+        BackgroundColor = UIColor.Orange
+};
+
+// This could be any parent view you add UI elements to.
+var rootView = RootViewController.View;
+rootView.AddSubview (orangeLabel);
+```
+
+#### Absolute coordinates
+
+Let’s position the `orangeLabel` at the top left corner manualy first so we can see it.
+
+```csharp
+orangeLabel.Frame = new CGRect (20, 20, 120, 50);
+```
+
+If we wanted to center this label in the middle of the screen we could, *technically*, do it using **absolute** x,y coordinates but this is **not** the most optimal, responsive and simple (will imply quite a bit of math) way to do it.
+
+#### Begin with Constaints
+
+`TranslatesAutoresizingMaskIntoConstraints` is **key** to using manual constraints.
+
+By default this is set to `true` which lets the system create a set of constraints for you.
+
+However it prevents you from adding **additional constraints**, see API doc:
+
+> Note that the autoresizing mask constraints fully specify the view’s size and position; therefore, you cannot add additional constraints to modify this size or position without introducing conflicts. If you want to use Auto Layout to dynamically calculate the size and position of your view, you must set this property to false, and then provide a non ambiguous, nonconflicting set of constraints for the view.
+
+```csharp
+orangeLabel.TranslatesAutoresizingMaskIntoConstraints = false;
+```
+
+Setting `TranslatesAutoresizingMaskIntoConstraints` to `false` should result in the label disappearing because we did not, *yet*, add our manual constraints.
```

Comment:
This is a bug in workbook, it runs all the code in the workbook, always. See: https://trello.com/c/tfzeDla6/913-cell-play-button-label-is-misleading-button-possibly-behaving-incorrectly.

Technically it should work so I don't really want to change my sentence here since it would go against the idea of workbooks (let's not have people try it in a classic app project, workbooks is the place to try and experiment).

It's on the workbooks team (: